packages/amplify-category-api/src/commands/api/generate-schema.ts

```diff
@@ -0,0 +1,80 @@
+import { $TSAny, $TSContext } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import * as path from 'path';
+import fs from 'fs-extra';
+import _ from 'lodash';
+import { MySQLDataSourceAdapter, generateGraphQLSchema, Schema, Engine, DataSourceAdapter } from '@aws-amplify/graphql-schema-generator';
+import { getDBUserSecretsWalkthrough } from '../../provider-utils/awscloudformation/service-walkthroughs/generate-graphql-schema-walkthrough';
+import { ImportedRDSType, RDS_SCHEMA_FILE_NAME } from '../../provider-utils/awscloudformation/service-walkthrough-types/import-appsync-api-types';
+import { readGlobalAmplifyInput, validateInputConfig } from '../../provider-utils/awscloudformation/utils/import-rds-utils/globalAmplifyInputs';
+import { getAppSyncAPIName, getAPIResourceDir } from '../../provider-utils/awscloudformation/utils/amplify-meta-utils';
+import { getExistingConnectionSecrets, storeConnectionSecrets } from '../../provider-utils/awscloudformation/utils/rds-secrets/database-secrets';
+import { writeSchemaFile } from '../../provider-utils/awscloudformation/utils/graphql-schema-utils';
+
+const subcommand = 'generate-schema';
+
+export const name = subcommand;
+
+export const run = async (context: $TSContext) => {
+  const apiName = getAppSyncAPIName();
+  const apiResourceDir = getAPIResourceDir(apiName);
+
+  // proceed if there are any existing imported Relational Data Sources
+  const pathToSchemaFile = path.join(apiResourceDir, RDS_SCHEMA_FILE_NAME);
+  if(fs.existsSync(pathToSchemaFile)) {
+    // read and validate the RDS connection parameters
+    const config: $TSAny = await readGlobalAmplifyInput(context, pathToSchemaFile);
+    // ensure that the required database connection details exist
+    await validateInputConfig(context, config);
+
+    // read and validate the RDS connection secrets
+    let secretsExistInParameterStore = true;
+    let secrets = await getExistingConnectionSecrets(context, config.database);
+    if(!secrets) {
+      secrets = await getDBUserSecretsWalkthrough(context, config.database);
+      secretsExistInParameterStore = false;
+    }
```

Comment:
The code smell here is the use of `let`. There are times when a `let` is necessary, but I'd prefer smaller, atomic functions that can return when necessary, rather than iteratively mutating state, and suggests a good opportunity to pull some logic out into a smaller, easily understood, more easily tested function.

Another example of that down below is the adapter/schema assignment thing. As written, it requires the two lets, then null checks, etc. instead of returning from a `getDatabaseForConfig` method, or something like, that.

Reply:
Refactored this logic more in https://github.com/aws-amplify/amplify-category-api/pull/1176